### DISCUSSION TITLE
Subtree under x-if still evaluated when condition turns falsy

### DISCUSSION THREAD

**Discussion by ptsoccer** [alpinejs/alpine#1926] [1. Help] [answered]
I think this is similar to https://github.com/alpinejs/alpine/discussions/1900. When there is a template with an x-if whose conditon is true and eventually turns falsy, the bindings in the subtree of it will be re-evaluated, potentially causing console errors if the bindings aren't written in a certain way to account for this.  [This is a boiled down version of the issue](https://jsfiddle.net/qayedv7k/)

Mainly I'm asking if this is the expected behavior or not. Obviously, I could write the bindings to account for null values, but I was kind of expecting the difference between x-if and x-show to be that x-show would keep re-evaluating bindings after a falsy condition, and x-if wouldn't.

**Comment by SimoTod** [comment]
~Are you using the latest version of Alpine? I believe this bug has been fixed in v3.2.3
(As a separate note, the demo works correctly for me).~

**Reply by ptsoccer** [reply]
I'm using https://unpkg.com/alpinejs to include the script which redirects to https://unpkg.com/alpinejs@3.2.3/dist/cdn.min.js

The demo works, but that's because I'm not doing anything complex in the bindings that could cause side effects

**Reply by SimoTod** [reply]
Scratch that, it seems it's still logging an error

**Comment by kylethielk** [comment, marked as answer]
hey @ptsoccer there are a couple of different reports that all seem to boil down to this same underlying cause. In V2 if x-if evaluated to a falsy value, anything contained within the x-if was not evaluated. In V3 this has changed. See #1917 and #1837 

I believe we still need a determination on whether this is intended or unintended.

**Reply by ptsoccer** [reply]
Seems reasonable, thank you
